router/batchrouter/batchrouter.go

```diff
@@ -1129,11 +1130,18 @@ func (brt *HandleT) postToWarehouse(batchJobs *BatchJobsT, output StorageUploadO
 		bytes.NewBuffer(jsonPayload))
 	if err != nil {
 		brt.logger.Errorf("BRT: Failed to route staging file URL to warehouse service@%v, error:%v", uri, err)
-	} else {
-		brt.logger.Infof("BRT: Routed successfully staging file URL to warehouse service@%v", uri)
-		defer resp.Body.Close()
+		return
 	}
+	defer func() { _ = resp.Body.Close() }()
 
+	if resp.StatusCode == http.StatusOK {
+		_, err = io.Copy(io.Discard, resp.Body)
```

Comment:
why do we need this?

Reply:
https://gist.github.com/mholt/eba0f2cc96658be0f717